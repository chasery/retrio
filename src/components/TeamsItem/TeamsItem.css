.TeamsItem__role {
  margin-top: 0.5rem;
}
.TeamsItem__email {
  font-weight: normal;
}
.TeamsItem__controls > * {
  margin-right: 0.5rem;
}
