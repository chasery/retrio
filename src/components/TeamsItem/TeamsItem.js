import React from 'react';
import { Link } from 'react-router-dom';
import './TeamsItem.css';

function TeamsItem(props) {
  const { id, name, owner } = props;

  return (
    <li className='TeamsItem'>
      <div className='TeamsItem__info'>
        <h3 className='TeamsItem__name'>{name}</h3>
        <div className='TeamsItem__role'>{owner ? 'Owner' : 'Contributor'}</div>
      </div>
      <ul className='TeamsItem__controls'>
        <li>
          <Link className='TeamsItem__button primary' to={`/teams/${id}`}>
            Manage
          </Link>
        </li>
      </ul>
    </li>
  );
}

export default TeamsItem;
