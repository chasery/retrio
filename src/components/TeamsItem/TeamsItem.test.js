import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TeamsItem from './TeamsItem';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
  useParams: () => ({
    id: '',
  }),
}));

describe('TeamsItem component', () => {
  it('renders without error', () => {
    const wrapper = shallow(<TeamsItem />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
