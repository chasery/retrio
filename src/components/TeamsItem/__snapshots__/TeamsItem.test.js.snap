// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamsItem component renders without error 1`] = `
<li
  className="TeamsItem"
>
  <div
    className="TeamsItem__info"
  >
    <h3
      className="TeamsItem__name"
    />
    <div
      className="TeamsItem__role"
    >
      Contributor
    </div>
  </div>
  <ul
    className="TeamsItem__controls"
  >
    <li>
      <Link
        className="TeamsItem__button primary"
        to="/teams/undefined"
      >
        Manage
      </Link>
    </li>
  </ul>
</li>
`;
